modules/table/src/main/java/org/apache/ignite/internal/table/distributed/index/IndexMetaStorage.java

```diff
@@ -360,18 +365,27 @@ private CompletableFuture<Void> recoverIndexMetas() {
             }
 
             // Let's identify the indexes that were removed from the catalog in this version.
+            Set<Integer> knownRemovedIndexIds = new HashSet<>();
```

Comment:
Why do we need this intermediate set? Can we add elements to `indexIds` directly?

Reply:
We can, but this way it makes it more difficult to understand the code because `indexIds` collected before the internal loop only contains 'from-catalog' indices, but now we suddenly add 'not-from-catalog' indices to them. This looks pretty confusing even for me who just grokked the code; for others it will be even more confusing